### PR TITLE
TIP-842: Extract variant attribute set interface to Akeneo\Pim\Structure

### PR DESCRIPTION
**Pull request by arnolanglade**
**Description (for Contributor and Core Developer)**

They are the last classes we did not moved from to catalog to the new BC context "Structure": Attribute and attribute translation.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | yes
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Component/spec/Model/FamilyVariantSpec.php [hunk 1]
```diff
@@ -158,7 +158,7 @@ function it_gets_the_attribute_set_level_for_provided_attribute_code()
         $name = new Attribute();
         $name->setCode('name');
 
-        $variantAttributeSet = new VariantAttributeSet();
+        $variantAttributeSet = new \Akeneo\Pim\Structure\Component\Model\VariantAttributeSet();
```
Comment (jjanvier) [inline src/Akeneo/Pim/Structure/Component/spec/Model/FamilyVariantSpec.php]: revert

File: src/Akeneo/Pim/Structure/Component/spec/Model/FamilyVariantSpec.php [hunk 2]
```diff
@@ -195,7 +195,7 @@ function it_gets_the_attribute_set_level_for_provided_common_attribute_code()
...
```
Comment (jjanvier) [inline src/Akeneo/Pim/Structure/Component/spec/Model/FamilyVariantSpec.php]: revert

File: src/Akeneo/Pim/Structure/Component/spec/Model/VariantAttributeSetSpec.php [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 use Akeneo\Pim\Structure\Component\Model\AttributeInterface;
-use Pim\Component\Catalog\Model\VariantAttributeSet;
-use Pim\Component\Catalog\Model\VariantAttributeSetInterface;
+use Akeneo\Pim\Structure\Component\Model\VariantAttributeSet;
+use Akeneo\Pim\Structure\Component\Model\VariantAttributeSetInterface;
 
 class VariantAttributeSetSpec extends ObjectBehavior
 {
-    function it_is_initializable()
+    function it_is_initializale()
```
Comment (LaurentPetard) [inline src/Akeneo/Pim/Structure/Component/spec/Model/VariantAttributeSetSpec.php]: it_is_initializable ?

Comment (jjanvier) [reply src/Akeneo/Pim/Structure/Component/spec/Model/VariantAttributeSetSpec.php]: ah
thanks

**Pull request merged**